### PR TITLE
feat: Add the ability to add new automation rule 

### LINKED ISSUE #3417: feat: Add the ability to create a new automation rule
- Add the ability to create new automation using `automation_name`, `automation_description`, `automation_event`, `automation_conditions` and `automation_actions`

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes #3417

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+              id: i.id,
+              name: i.title,
+            };
+          });
+        case 'browser_language':
+          return languages;
+        case 'country_code':
+          return countries;
+        default:
+          break;
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue:267]: `return undefined` would fix the eslint error

File: app/javascript/dashboard/components/widgets/ConfirmModal.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <modal :show.sync="show" :on-close="onClose">
+    <div class="column content-box">
+      <woot-modal-header header-title="Change Event type"> </woot-modal-header>
+      <div class="row modal-content">
+        <div class="medium-12 columns">
+          <p>
+            Changing the event type will reset the condition and actions you
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/ConfirmModal.vue:8]: Use i18n here

File: app/javascript/dashboard/components/widgets/ConfirmModal.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            have added below.
+          </p>
+        </div>
+        <div class="modal-footer">
+          <div class="medium-12 columns">
+            <woot-button @click="confirm">
+              Proceed
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/ConfirmModal.vue:15]: Use i18n here also

File: app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+              >
+                {{ event.value }}
+              </option>
+            </select>
+            <span v-if="$v.automation.event_name.$error" class="message">
+              {{ $t('AUTOMATION.ADD.FORM.EVENT.ERROR') }}
+            </span>
+          </label>
+          <p v-if="hasAutomationMutated" class="info-message">
+            Changing event type will reset the conditions and events you have
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue:49]: Use i18n here

File: app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+.event_wrapper {
+  select {
+    margin: 0;
+  }
+  .info-message {
+    font-size: var(--font-size-mini);
+    color: #868686;
+    text-align: right;
+  }
+  margin-bottom: 1.6rem;
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue:443]: Use variables

**Comment by muhsin-k** [comment]
@fayazara The code looks good to me. Let's disable the automation route and merge this PR

**Comment by fayazara** [comment]
> @fayazara The code looks good to me. Let's disable the automation route and merge this PR

Disabled automation route @muhsin-k

File: app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue (hunk 5 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue:444]: @fayazara

File: app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+.filter-actions {
+  margin-top: var(--space-normal);
+}
+.event_wrapper {
+  select {
+    margin: 0;
+  }
+  .info-message {
+    font-size: var(--font-size-mini);
+    color: #868686;
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue:441]: Try using
```suggestion
    color: var(--b-500) or color: var(--b-600);
```

**Review by nithindavid — DISMISSED** [review]

**Comment by muhsin-k** [comment]
@fayazara Please resolve the conflicts

**Comment by muhsin-k** [comment]
@tejaswinichile Can we start QA for all the actions/conditions?

**Comment by fayazara** [comment]
> @fayazara Please resolve the conflicts

Fixed @muhsin-k

**Comment by tejaswinichile** [comment]
> @tejaswinichile Can we start QA for all the actions/conditions?

@muhsin-k I have done at my local, it's working fine(may be data issue), if we face any issues we can add the bug fix for the same.

**Review by fayazara — APPROVED** [review]

**Pull request merged**